core/src/main/java/com/linecorp/armeria/common/stream/PublisherBasedStreamMessage.java

```diff
@@ -204,21 +212,41 @@ private static void failLateSubscriber(EventExecutor executor,
 
     @Override
     public void abort() {
+        abort(AbortedStreamException::get);
+    }
+
+    @Override
+    public void abort(Throwable cause) {
+        requireNonNull(cause, "cause");
+        abort(() -> cause);
+    }
+
+    @Override
+    public void abort(Supplier<? extends Throwable> causeSupplier) {
+        requireNonNull(causeSupplier, "causeSupplier");
+        abort0(causeSupplier);
+    }
+
+    private void abort0(Supplier<? extends Throwable> causeSupplier) {
         final AbortableSubscriber subscriber = this.subscriber;
+        final Throwable cause = requireNonNull(causeSupplier.get(),
```

Comment:
What is the point of using `Supplier` if you’re going to call `get()` regardless of whether `abort()` will fail?